src/DashboardService.Data/PriorityRepository.cs

```diff
@@ -18,28 +21,83 @@ public PriorityRepository(IDataProvider provider)
     _provider = provider;
   }
   
-  public Task<Guid?> CreateAsync(DbPriority dbDepartment)
+  public async Task<Guid?> CreateAsync(DbPriority dbPriority)
   {
-    throw new NotImplementedException();
+    _provider.Priorities.Add(dbPriority);
+    await _provider.SaveAsync();
+
+    return dbPriority.Id;
+  }
+
+  public async Task<(List<DbPriority> dbPriorities, int totalCount)> GetAllAsync(GetPrioritiesFilter filter, CancellationToken cancellationToken)
+  {
+    IQueryable<DbPriority> dbPriorities = _provider.Priorities.AsQueryable();
```

Comment:
It is queryable. No need for AsQueryably(). You'd better use AsNoTracking()

Reply:
++